Source/WebKit/UIProcess/Cocoa/_WKWarningView.mm

```diff
@@ -180,71 +181,180 @@
         });
     case WarningItem::TitleText:
     case WarningItem::ExclamationPoint:
+    case WarningItem::BrokenLock:
         return WTF::switchOn(warningView.warning->data(), [&] (const WebKit::BrowsingWarning::SafeBrowsingWarningData&) {
             return narrow ? white : red;
         }, [&] (const WebKit::BrowsingWarning::HTTPSNavigationFailureData&) {
-            return !narrow && item == WarningItem::TitleText ? white : red;
+            return item == WarningItem::TitleText ? darkGray : red;
         });
     case WarningItem::MessageText:
     case WarningItem::ShowDetailsButton:
     case WarningItem::ContinueButton:
         return WTF::switchOn(warningView.warning->data(), [&] (const WebKit::BrowsingWarning::SafeBrowsingWarningData&) {
             return narrow ? white : [UIColor darkTextColor];
         }, [&] (const WebKit::BrowsingWarning::HTTPSNavigationFailureData&) {
-            return item == WarningItem::MessageText ? mediumGray : white;
+            return darkGray;
         });
     case WarningItem::GoBackButton:
         return WTF::switchOn(warningView.warning->data(), [&] (const WebKit::BrowsingWarning::SafeBrowsingWarningData&) {
             return narrow ? white : warning.tintColor;
         }, [&] (const WebKit::BrowsingWarning::HTTPSNavigationFailureData&) {
-            return white;
+            return narrow ? darkGray : warning.tintColor;
         });
     }
 #endif
     ASSERT_NOT_REACHED();
     return nil;
 }
 
-@interface WKSafeBrowsingExclamationPoint : ViewType
-@end
+#if PLATFORM(MAC)
+using BezierPath = NSBezierPath;
+#else
+using BezierPath = UIBezierPath;
+#endif
 
-@implementation WKSafeBrowsingExclamationPoint
+static void addLineToPoint(BezierPath *path, CGPoint point)
+{
+#if PLATFORM(MAC)
+    [path lineToPoint:point];
+#else
+    [path addLineToPoint:point];
+#endif
+}
 
-- (void)drawRect:(RectType)rect
+static void addCurveToPoint(BezierPath *path, CGPoint endPoint, CGPoint controlPoint1, CGPoint controlPoint2)
+{
+#if PLATFORM(MAC)
+    [path curveToPoint:endPoint controlPoint1:controlPoint1 controlPoint2:controlPoint2];
+#else
+    [path addCurveToPoint:endPoint controlPoint1:controlPoint1 controlPoint2:controlPoint2];
+#endif
+}
+
+static void addLockIconPath(BezierPath *path)
```

Comment:
Can we use an SF symbol (and `UIImageView` / `NSImageView`)? Based on the description it seems like this would be `lock.slash.fill`.

Reply:
Thanks! `lock.slash.fill` was exactly what I wanted. I'm also migrating the exclamation mark to use the SF symbol.